app/services/spaces/duplicate_detector_service.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module Spaces
+  class DuplicateDetectorService < ApplicationService
+    def initialize(space)
+      @title = space.title
+      @address = space.address
+      @post_number = space.post_number
+      super()
+    end
```

Comment:
This service looks pretty good although convention in the ruby world is to add attr_readers to each of the fields so 
`attr_reader :title, :address, :space` a proper guide for this here https://www.rubyguides.com/2018/11/attr_accessor/

Reply:
Would that also be needed when I don't want to expose them to the outside world? These aren't used by any external resources calling the service.

Reply:
Fixed!